### DISCUSSION TITLE
Enum from JS

### DISCUSSION THREAD

**Discussion by lamazze** [sebastienros/jint#1478] [Q&A] [answered]
Hello all,

Probably something obvious but I have a problem to use an c# enum from Jint.

I have a simple class:

```
class MyClass : IDisposable
{
	public enum e { one, two, three };
	public const string myString = "test";
	public void MyMethod()...
}
```
I share a instantiation of that class with the Jint Engine.
I can use myString and MyMethod() with no problem but not the enum.
A calll to class.e raise an exception: TypeError: e is undefined

What am I doing wrong ?

Thank you !
BR,
lamaZZe

**Comment by lahma** [comment, marked as answer]
Jint cannot see type `MyClass.e` as it's a nested type, it only properties can be accessed this way. You can create explicit type reference if needed, see README:

```c#
engine.SetValue("TheType", TypeReference.CreateTypeReference(engine, typeof(TheType)))
```

**Comment by lamazze** [comment]
Thank you
